### ISSUE TITLE
Usage with loadable-componets

### ISSUE THREAD

**Issue by yyynnn** [stereobooster/react-snap#299]
Hi there!

So i'm trying to use `react-snap` wo SSR to prerender SPA and i have some `loadable-components`.
Been using this entry:

```javascript
...
import { loadableReady } from '@loadable/component'

const Render = () => (
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <ConnectedRouter history={history}>
        <App />
      </ConnectedRouter>
    </PersistGate>
  </Provider>
)

const rootElement = document.getElementById('app')

if (rootElement.hasChildNodes()) {
  loadableReady().then(() => {
    hydrate(<Render />, rootElement)
  })
} else {
  render(<Render />, rootElement)
}

```
App loads fine, but post-build with `react-snap` results in this error:
`loadableReady() requires state, please use getScriptTags or getScriptElements server-side`

cant i just pre render components client side?

**Comment by stereobooster** [comment]
The issue is that latest `loadable-components` doesn't provide `getState()` (#297). Without it you can't use `loadableReady()`. So you can either downgrade `loadable-components` and use `getState()` or remove `loadableReady()` (which will lead to flash).

**Comment by yyynnn** [comment]
Yes, but when i'm using 2.2.3 with `getState()` i get `loadable-components state not found. You have a problem server-side. Please verify that you have called `loadableState.getScriptTag()` server-side.` error.
without `loadableReady()` at all, just with regular `render(...)` i get `pageerror at /private/page/uuid: SyntaxError: Unexpected token <` at path with loadable component.

loadable generator:

```javascript

export const PageGenerator = ({ data }: Props) => {
  return (
    <React.Fragment>
      {data.map(({ name, id }: ComponentProps, i) => {
        const LoadableComponent = loadable(
          () => import(`lazies/${name}.lazy.ts`),
          {
            modules: [''] // Should this be used at all?
          }
        )
        return (
          <React.Fragment key={i}>
            <LoadableComponent id={id} />
          </React.Fragment>
        )
      })}
    </React.Fragment>
  )
}
```

**Comment by yyynnn** [comment]
Ok, now it falis at loadable component `pageerror at /private/page/uuid: SyntaxError: Unexpected token <`

**Comment by stereobooster** [comment]
well. It is hard to be sure without seeing the code, but I guess it tries to load js file from the server, which in this case not js, but ts. There is no such file `lazies/abc.lazy.ts`, because there is `lazies/abc.lazy.js` as the result server return html instead of JS and this why it says `SyntaxError: Unexpected token <`

**Comment by stereobooster** [comment]
Reproducible demo, which I can download, run locally and debug

**Comment by stereobooster** [comment]
You can try to reproduce this error with create react app 2.1, which supports TS. At least you can crosspost to their tracker maybe they will help you

**Comment by stereobooster** [comment]
I reported issue with Suspense to React https://github.com/facebook/react/issues/14438. Let's see what they would suggest.
